### PR TITLE
Feature infinite scroll

### PR DESCRIPTION
**Pull request by philippe-cleany**

### CODE REVIEW

**Review by thomas-cleany — APPROVED** [review]

File: src/reducer/resource/list/ids.js [hunk 1]
```diff
@@ -5,8 +5,17 @@ export default resource => (previousState = [], { type, payload, requestPayload,
...
     }
     switch (type) {
-    case CRUD_GET_LIST_SUCCESS:
-        return payload.data.map(record => record.id);
+    case CRUD_GET_LIST_SUCCESS:{
+        if (meta.infiniteList) {
+          if (requestPayload.pagination.page === 1) {
+            return payload.data.map(record => record.id);
+          }
+          return previousState.concat(payload.data.map(record => record.id));
```
Comment (sebastien-cleany) [inline src/reducer/resource/list/ids.js]: if (meta.infiniteList || requestPayload.pagination.page === 1) {
  return payload.data.map(record => record.id);
}
return previousState.concat(payload.data.map(record => record.id));

File: src/mui/list/InfiniteList.js [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+import DefaultPagination from './Pagination';
+import DefaultActions from './Actions';
+import { crudGetList as crudGetListAction } from '../../actions/dataActions';
+import { changeListParams as changeListParamsAction } from '../../actions/listActions';
+import translate from '../../i18n/translate';
+import removeKey from '../../util/removeKey';
+import defaultTheme from '../defaultTheme';
+import InfiniteScroll from 'react-infinite-scroller';
+import LinearProgress from 'material-ui/LinearProgress';
+import { List } from './List';
```
Comment (sebastien-cleany) [inline src/mui/list/InfiniteList.js:23]: Is all this include needed ?

File: src/mui/list/InfiniteList.js [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+const styles = {
+    noResults: { padding: 20 },
+    header: {
+        display: 'flex',
+        justifyContent: 'space-between',
+    },
+};
+
+export class InfiniteList extends List {
+  constructor(props, state) {
```
Comment (sebastien-cleany) [inline src/mui/list/InfiniteList.js]: useless constructor

File: src/mui/list/InfiniteList.js [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+
+  componentDidMount() {
+      this.updateData();
+      if (Object.keys(this.props.query).length > 0) {
+          this.props.query.page = 0;
+          this.props.changeListParams(this.props.resource, this.props.query);
+      }
+  }
+
+  getQuery() {
```
Comment (sebastien-cleany) [inline src/mui/list/InfiniteList.js:44]: const query = super.getQuery()
query.page = this.props.params.page;
return query;

File: src/mui/list/InfiniteList.js [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+          query.sort = this.props.sort.field;
+          query.order = this.props.sort.order;
+      }
+      if (!query.perPage) {
+          query.perPage = this.props.perPage;
+      }
+      return query;
+  }
+
+  updateData(query) {
```
Comment (sebastien-cleany) [inline src/mui/list/InfiniteList.js:57]: Is dup needed ?

File: src/mui/list/InfiniteList.js [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+      const params = query || this.getQuery();
+      const { sort, order, page, perPage, filter } = params;
+      const pagination = { page: parseInt(page, 10), perPage: parseInt(perPage, 10) };
+      const permanentFilter = this.props.filter;
+      this.props.crudGetList(this.props.resource, pagination, { field: sort, order }, { ...filter, ...permanentFilter }, false, true);
+  }
+
+  getNextPage() {
+       this.changeParams({type: 'INC_PAGE'})
```
Comment (sebastien-cleany) [inline src/mui/list/InfiniteList.js]: weird indent

File: src/mui/list/InfiniteList.js [hunk 7]
```diff
@@ -0,0 +1,182 @@
...
+                  }
+                  </InfiniteScroll>
+              </Card>
+          </div>
+      );
+  }
+}
+
+const getLocationSearch = props => props.location.search;
+const getQuery = createSelector(
```
Comment (sebastien-cleany) [inline src/mui/list/InfiniteList.js:140]: Useless ?

File: src/mui/list/InfiniteList.js [hunk 8]
```diff
@@ -0,0 +1,182 @@
...
+                    <CardText style={styles.noResults}>{translate('aor.navigation.no_results')}</CardText>
+                  }
+                  </InfiniteScroll>
+              </Card>
+          </div>
+      );
+  }
+}
+
+const getLocationSearch = props => props.location.search;
```
Comment (sebastien-cleany) [inline src/mui/list/InfiniteList.js:139]: useless ?

File: src/mui/list/InfiniteList.js [hunk 9]
```diff
@@ -0,0 +1,182 @@
...
+        const query = parse(locationSearch);
+        if (query.filter && typeof query.filter === 'string') {
+            query.filter = JSON.parse(query.filter);
+        }
+        return query;
+    },
+);
+
+function mapStateToProps(state, props) {
+    const resourceState = state.admin[props.resource];
```
Comment (sebastien-cleany) [inline src/mui/list/InfiniteList.js:152]: const resourceState = state.admin[props.resource];
const newProps = ListMapStateToProps(state, props);
newProps.hasMore: resourceState.list.ids.length < resourceState.list.total;
return newProps

File: src/mui/list/InfiniteList.spec.js [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+                children={[]}
+                total={0}
+                changeFormValue={() => true}
+                changeListParams={() => true}
+            >
+                <div />
+            </InfiniteList>
+        );
+        const textElement = wrapper.find('CardText').children();
+        console.log('%c [InfiniteList]', 'color:red;',textElement.text());
```
Comment (sebastien-cleany) [inline src/mui/list/InfiniteList.spec.js]: console

File: src/mui/list/InfiniteList.js (hunk 2 above)
Comment (philippe-cleany) [reply src/mui/list/InfiniteList.js:23]: absolutely !

File: src/mui/list/InfiniteList.js (hunk 3 above)
Comment (philippe-cleany) [reply src/mui/list/InfiniteList.js]: ok

File: src/mui/list/InfiniteList.js (hunk 7 above)
Comment (philippe-cleany) [reply src/mui/list/InfiniteList.js:140]: nope !

File: src/mui/list/InfiniteList.js (hunk 8 above)
Comment (philippe-cleany) [reply src/mui/list/InfiniteList.js:139]: nope !

File: src/mui/list/InfiniteList.js (hunk 5 above)
Comment (philippe-cleany) [reply src/mui/list/InfiniteList.js:57]: Yes because we set GrudGetList with diff parameters

File: src/mui/list/InfiniteList.js (hunk 4 above)
Comment (philippe-cleany) [reply src/mui/list/InfiniteList.js:44]: ok

File: src/mui/list/InfiniteList.js (hunk 3 above)
Comment (philippe-cleany) [reply src/mui/list/InfiniteList.js]: ok

File: src/reducer/resource/list/ids.js (hunk 1 above)
Comment (philippe-cleany) [reply src/reducer/resource/list/ids.js]: it is not this behavior that we want

File: src/mui/list/InfiniteList.js (hunk 6 above)
Comment (philippe-cleany) [reply src/mui/list/InfiniteList.js]: ok

**Pull request closed without merging**